display/display.go

```diff
@@ -246,6 +252,89 @@ func Init(version, commit, builtBy string) {
 		// Other potential keys are Tab and Backtab, they are ignored
 	})
 
+	searchBar.SetDoneFunc(func(key tcell.Key) {
+		tab := curTab
+
+		reset := func() {
+			searchBar.SetLabel("")
+			tabs[tab].applyAll()
+			App.SetFocus(tabs[tab].view)
+			layout.RemoveItem(searchBar)
+			tabs[tab].mode = tabModeDone
+		}
+
+		//nolint:exhaustive
+		switch key {
+		case tcell.KeyEnter:
+			// Figure out whether it's a URL, link number, or search
+			// And send out a request
+
+			// Escape the search string to not find regexp symbols
+			searchString = regexp.QuoteMeta(searchBar.GetText())
+
+			if strings.TrimSpace(searchString) == "" {
+				// Ignore
+				reset()
+				return
+			}
+
+			if tabs[tab].mode != tabModeSearch {
+				originalText = tabs[curTab].view.GetBytes(false)
+			}
+			tabs[tab].mode = tabModeSearch
+
+			// find all positions of the search string
+			searchRegex := regexp.MustCompile(searchString)
+			searchIdx := searchRegex.FindAllIndex(originalText, -1)
+
+			// find all positions of tags
+			tagsRegex := regexp.MustCompile(`\[.*?[^\[]\]`)
+			tagsIdx := tagsRegex.FindAllIndex(originalText, -1)
+
+			text := []byte("")
+			matches = 0
+			lastMatch := 0
+			var isMatch bool
+
+			// loops through all occurrences and check if they
+			// discard if they lie within tags.
+			// []byte text is build from the original text buffer
+			// with the actual search strings replaced by tagged regions
+			// to highlight.
+			for i, match := range searchIdx {
+				isMatch = true
```

Comment:
Isn't this variable unnecessary? You already break out of the loop if it's false, so the check on 313 isn't needed either.